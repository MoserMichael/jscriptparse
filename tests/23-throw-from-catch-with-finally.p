# checks formatting of error messages

def rec(num) {
    if num == 0 {
      throw "throwing an error string"
    }
    return rec(num - 1)
}        

def tryIt() {
    try 
        rec(5)
    catch ex 
      println("handling exception")
    finally 
      throw "throw from finally"
}

tryIt()
