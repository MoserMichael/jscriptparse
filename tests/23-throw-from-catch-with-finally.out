handling exception
Error: "throw from finally"
#(tests/23-throw-from-catch-with-finally.p:3)       throw "throw from finally"
                                            |.......^
#(tests/23-throw-from-catch-with-finally.p:2) def tryIt() {
                                            |.^
#(tests/23-throw-from-catch-with-finally.p:1) tryIt()
                                            |.^
